src/common/snippets/src/pass/insert_movebroadcast.cpp

```diff
@@ -118,3 +119,25 @@ ngraph::snippets::pass::InsertMoveBroadcast::InsertMoveBroadcast() {
 
     register_matcher(std::make_shared<ngraph::pattern::Matcher>(any, matcher_name), callback);
 }
+
+ngraph::snippets::pass::BroadcastToMoveBroadcast::BroadcastToMoveBroadcast() {
+    MATCHER_SCOPE(BroadcastToMoveBroadcast);
+
+    register_matcher(std::make_shared<ngraph::pattern::Matcher>(ngraph::pattern::wrap_type<ngraph::opset1::Broadcast>(), matcher_name),
+            [this](ngraph::pattern::Matcher &m) {
+        OV_ITT_SCOPED_TASK(ngraph::pass::itt::domains::SnippetsTransform, "Snippets::op::BroadcastToMoveBroadcast")
+        auto root = m.get_match_root();
+        auto broadcast = ov::as_type_ptr<ngraph::opset1::Broadcast>(root);
+        if (broadcast->get_broadcast_spec() != ngraph::op::AutoBroadcastType::NUMPY) {
+            return false;
+        }
+
+        auto broadcast_move = broadcast_node_last_dim(broadcast->input_value(0), broadcast->get_output_shape(0), broadcast->get_input_shape(0));
+        auto target_output = ov::is_type<ngraph::snippets::op::BroadcastMove>(broadcast_move) ? broadcast_move->output(0) :
```

Comment:
Why outdated?) You can still use `broadcast_node_last_dim` from another file to avoid code duplication, just make it a `static` method. It makes perfect sense to reuse the `broadcast_node_last_dim` logic from another transformation to avoid code duplication and to keep the transformations consistent.

Reply:
Right. Done, thanks a lot!